staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go

```diff
@@ -54,15 +54,12 @@ func UpdateResource(r rest.Updater, scope *RequestScope, admit admission.Interfa
 			return
 		}
 
-		// TODO: we either want to remove timeout or document it (if we document, move timeout out of this function and declare it in api_installer)
-		timeout := parseTimeout(req.URL.Query().Get("timeout"))
-
 		namespace, name, err := scope.Namer.Name(req)
 		if err != nil {
 			scope.err(err, w, req)
 			return
 		}
-		ctx, cancel := context.WithTimeout(req.Context(), timeout)
+		ctx, cancel := request.WithSecondaryDeadline(req.Context(), requestTimeout)
```

Comment:
WithShorterDeadline?

Reply:
I removed `WithSecondaryDeadline` as we no longer need it.   https://github.com/kubernetes/kubernetes/pull/96061#discussion_r520021733